homeassistant/components/media_player/sonos.py

```diff
@@ -426,7 +426,17 @@ def _set_basic_information(self):
         self._play_mode = self.soco.play_mode
         self._night_sound = self.soco.night_mode
         self._speech_enhance = self.soco.dialog_mode
-        self._favorites = self.soco.music_library.get_sonos_favorites()
+
+        self._favorites = []
+        for fav in self.soco.music_library.get_sonos_favorites():
+            # SoCo 0.14 raises a generic Exception on invalid xml in favorites.
+            # Filter those out now so our list is safe to use.
+            try:
+                if fav.reference.get_uri():
+                    self._favorites.append(fav)
+            # pylint: disable=broad-except
+            except Exception:
```

Comment:
Can we set the core exception for soco?

Reply:
Unfortunately, SoCo raises `Exception` in this situation.

I already reported this upstream and it should be fixed in their next release: https://github.com/SoCo/SoCo/pull/601/files